src/TalisOrm/AggregateRepository.php

```diff
@@ -237,4 +248,61 @@ private function select($selectExpression, $tableExpression, array $where)
 
         return $this->connection->executeQuery($sql, $values);
     }
+
+    private function rememberEntity(Entity $entity)
+    {
+        $this->knownEntities[spl_object_hash($entity)] = $entity;
```

Comment:
apparently `spl_object_hash` is subject to collsions (https://github.com/doctrine/mongodb-odm/issues/1017, http://php.net/manual/en/function.spl-object-hash.php) even tho I never ran into it.

An alternative could be http://php.net/manual/en/splobjectstorage.contains.php

Reply:
Also, `knownEntities` grows in memory and cannot be garbage collected without a manual `clear()` call.
That's a big shift in memory management for the API user.

Reply:
Good points! Didn't know about the hash collisions - nasty situation. I thought about maybe only keeping the identifiers, not the whole object reference. Then memory wouldn't be that much of an issue anymore.

Reply:
At first I thought, maybe let the user tell us if an entity is new or should be updated. But I think there isn't a good way for them to know. Except that they could do something like toggle an internal boolean flag once `getState()` is called. And do the same thing in `fromState()`. But it would introduce more work in userland code.

Reply:
With all these issues, I'm thinking, maybe it's still worth it to do that. Keep true to the spirit of this ORM and just let the user deal with this. From the start, the idea was: if you just let the user handle this, the ORM can be much much simpler. And now it's going to be more complicated by the minute hehe.

Reply:
I would be all for using the identifier only. I once talked to someone who had ran into the spl_hash collision problem and it's a very nasty one to get out of.

The only thing is you might run into trouble with child entities; when I remove a child entity and add it again with the same ID but different values the code will not detect that it changed when it looks at IDs only.